src/main/java/org/jabref/gui/bibtexextractor/BibtexExtractorViewModel.java

```diff
@@ -1,32 +1,99 @@
 package org.jabref.gui.bibtexextractor;
 
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
+import javafx.application.Platform;
 import javafx.beans.property.SimpleStringProperty;
 import javafx.beans.property.StringProperty;
+import javafx.concurrent.Task;
 
 import org.jabref.Globals;
+import org.jabref.JabRefGUI;
+import org.jabref.gui.DialogService;
+import org.jabref.logic.bibtexkeypattern.BibtexKeyGenerator;
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.logic.importer.fetcher.GrobidCitationFetcher;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.Defaults;
 import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.database.BibDatabaseMode;
 import org.jabref.model.entry.BibEntry;
 import org.jabref.model.entry.types.EntryType;
 import org.jabref.model.entry.types.StandardEntryType;
+import org.jabref.preferences.JabRefPreferences;
 
 public class BibtexExtractorViewModel {
 
     private final StringProperty inputTextProperty = new SimpleStringProperty("");
     private final BibDatabaseContext bibdatabaseContext;
+    private List<BibEntry> extractedEntries;
+    private final BibDatabaseContext newDatabaseContext;
+    private boolean directAdd;
+    private DialogService dialogService;
+    private GrobidCitationFetcher currentCitationfetcher;
 
     public BibtexExtractorViewModel(BibDatabaseContext bibdatabaseContext) {
         this.bibdatabaseContext = bibdatabaseContext;
+        newDatabaseContext = new BibDatabaseContext(new Defaults(BibDatabaseMode.BIBTEX));
+        dialogService = JabRefGUI.getMainFrame().getDialogService();
     }
 
     public StringProperty inputTextProperty() {
         return this.inputTextProperty;
     }
 
-    public void startExtraction() {
+    public void startParsing(boolean directAdd) {
+        this.directAdd = directAdd;
+        this.extractedEntries = null;
+        Task<Void> parseUsingGrobid = new Task<Void>() {
+            @Override
+            protected Void call() throws Exception {
+                try {
+                    currentCitationfetcher = new GrobidCitationFetcher(
```

Comment:
I think it would be better to initialize the fetcher in the constructor. 

Please also pass the necessary parameter (file monitor, preferences) as dependencies via constructor arguments. We are trying to get rid of `Globals`.